githubPort configuration setting is reported as not recognised in docusaurus.config.js

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Trying to execute the npm deploy command with the setting githubPort = '7999', results in an error:
[ERROR] Error: These field(s) ("githubPort",) are not recognized in docusaurus.config.js.
If you still want these fields to be in your configuration, put them in the "customFields" field.
See https://docusaurus.io/docs/api/docusaurus-config/#customfields
    at validateConfig (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/configValidation.js:201:15)
    at loadSiteConfig (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/config.js:25:62)
    at async loadContext (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/index.js:31:63)
    at async build (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/commands/build.js:50:21)


### Reproducible demo

_No response_

### Steps to reproduce

1. add a githubPort = '7999', option to docusaurus.config.js
2. run the deploy command


### Expected behavior

1. add a githubPort = '7999', option to docusaurus.config.js
2. run the deploy command
3. Deployment should run without an error

### Actual behavior


> docs@0.0.0 build
> docusaurus build

[ERROR] Error: These field(s) ("githubPort",) are not recognized in docusaurus.config.js.
If you still want these fields to be in your configuration, put them in the "customFields" field.
See https://docusaurus.io/docs/api/docusaurus-config/#customfields
    at validateConfig (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/configValidation.js:201:15)
    at loadSiteConfig (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/config.js:25:62)
    at async loadContext (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/server/index.js:31:63)
    at async build (/home/igs03102/sme/docs/node_modules/@docusaurus/core/lib/commands/build.js:50:21)

### Your environment

_No response_

### Self-service

- [X] I'd be willing to fix this bug myself.

--- a/packages/docusaurus/src/server/__tests__/configValidation.test.ts
+++ b/packages/docusaurus/src/server/__tests__/configValidation.test.ts
@@ -37,6 +37,7 @@ describe('normalizeConfig', () => {
       organizationName: 'facebook',
       projectName: 'docusaurus',
       githubHost: 'github.com',
+      githubPort: '8000',
       customFields: {
         myCustomField: '42',
       },

--- a/packages/docusaurus/src/server/configValidation.ts
+++ b/packages/docusaurus/src/server/configValidation.ts
@@ -185,6 +185,7 @@ export const ConfigSchema = Joi.object({
   deploymentBranch: Joi.string().optional(),
   customFields: Joi.object().unknown().default(DEFAULT_CONFIG.customFields),
   githubHost: Joi.string(),
+  githubPort: Joi.string(),
   plugins: Joi.array().items(PluginSchema).default(DEFAULT_CONFIG.plugins),
   themes: Joi.array().items(ThemeSchema).default(DEFAULT_CONFIG.themes),
   presets: Joi.array().items(PresetSchema).default(DEFAULT_CONFIG.presets),

